### ISSUE TITLE
Skin extending only works for templates, not for assets

### ISSUE THREAD

**Issue by rcubetrac** [roundcube/roundcubemail#5115]
_Reported by mgrum on 1 Aug 2014 10:03 UTC as Trac ticket #1490013_

The documentation at [says that using the skin extending mechanism "basically adds the base skin folder to the search path for templates and assets". However, in reality this only seems to work for templates, not for assets like CSS files and images. I think this really limits the usefulness of the extending mechanism.

Fortunately, this is really easy to fix:

```
@@ -658,7 +658,8 @@ class rcmail_output_html extends rcmail_output

         // correct absolute paths
         if ($file[0](http://trac.roundcube.net/wiki/Doc_Skins#Extendingskins]) == '/') {
-            $file = $this->base_path . $file;
+            $this->get_skin_file($file, $skin_path);
+            $file = $skin_path . $file;
         }

         // add file modification timestamp
```

_Migrated-From: http://trac.roundcube.net/ticket/1490013_

**Comment by dsoares** [comment]
This fix does not seem right. In `function file_callback`, variable `$skin_path` is unknown in the function scope. I noticed because i use another skin that extends larry and this commit completely messed up with the stylesheets.

**Comment by alecpl** [comment]
There was a followup fix in 3f53acb64a. Did you see it?

**Comment by dsoares** [comment]
Hi,
I saw it after i commented but still no luck with stylesheets (templates and images are OK).
Please test extending the skin larry like this and see if it works:

`skins/xlarry/meta.json`
```json
{
	"extends": "larry",
	"name": "Xlarry",
	"author": "test",
	"license": "Creative Commons Attribution-ShareAlike",
	"license-url": "http://creativecommons.org/licenses/by-sa/3.0/"
}
```

`skins/xlarry/includes/links.html`
```html
<roundcube:include file="/includes/links.html" skinPath="skins/larry" />
<link rel="stylesheet" type="text/css" href="/styles.css" />
```

`skins/xlarry/styles.css`
```css
#taskbar a {color:red}
```

**Comment by alecpl** [comment]
I reverted the change as the regression wasn't nice. I'm not sure we'll be able to have this request implemented keeping backward compatibility. This might be tricky, but I'll see what I can do.

**Comment by alecpl** [comment]
It should be ok now.

**Comment by dsoares** [comment]
Yes, it's impec! Thank you.

**Comment by arekm** [comment]
Can we have fix backported to 1.3?
